### PR TITLE
Load hardware config type hints

### PR DESCRIPTION
**Pull request by fall-byrd**
## TL;DR
Made a number of relatively minor changes along the backend load hardware config path. Highlights include: added type hints, removed unused imports, removed a duplicate property definition, (I believe) fixed the simulated thermometer getTemp value when not set in config, and other smaller changes.

## What
What is affected by this PR?
- [ ] API
- [ ] GUI
- [ ] Hardware Integration
- [ ] OS/Deployment
- [ ] Documentation
- [X] Other (please describe in notes)

## Testing
How was this tested?
- [ ] Locally Virtualized
- [ ] Raspberry Pi
- [ ] With Hardware
- [X] Other (please describe in notes)

## Notes
As before ran unit tests, ran the server and GUI against the running server. I'll add some notes in the PR for the motivation/reasoning behind some of the changes where it might be helpful (apologies if I explain something already known).

Also happy to commit, I love the ethos behind the org's group of projects and am very happy something like this exists at all.

### CODE REVIEW

File: backend/config.py [hunk 1]
```diff
@@ -63,10 +64,6 @@ def dataDirectory(self):
     def recipesDirectory(self):
         return '{0}/recipes/'.format(self.dataDirectory)
 
-    @property
-    def controllerHardwareDirectory(self):
```
Comment (fall-byrd) [inline backend/config.py:67]: This is defined again below just above the `labHardwareDirectory` property decleration.

File: backend/hardware/devicelist.py [hunk 2]
```diff
@@ -52,11 +50,11 @@ def setupDevices(deviceDefinitions):
     return devices
 
 
-def validateConfiguration(deviceConfigs):
+def validateConfiguration(deviceConfigs: list[dict]):
     deviceDict = {}
     for device in deviceConfigs:
 
-        if deviceDict.get(device['id'], None) == None:
+        if deviceDict.get(device['id'], None) is None:
```
Comment (fall-byrd) [inline backend/hardware/devicelist.py:57]: The comparison with `==` will definitely work but it is my understanding that the 'is' check is more encouraged now for `None` or `True` and `False`

File: backend/hardware/gpiochip/gpiod.py [hunk 3]
```diff
@@ -1,13 +1,13 @@
-from hardware.gpiochip.base import GPIOChip, LINE_REQ_DIR_OUT, LINE_REQ_DIR_IN
+from hardware.gpiochip.base import GPIOChip, LINE_REQ_DIR_OUT
 import gpiod
 import logging
 
 
 class GPIODChip(GPIOChip):
-    def __init__(self, args):
```
Comment (fall-byrd) [inline backend/hardware/gpiochip/gpiod.py:7]: I changed all the instances of `args` to a more appropriate config related value in each circumstance. Specifically in my experience the usage of `args` in method definitions is usually as a pass through of all argument and key name arguments when calling a parent class via something like `super()` ex:

``` python
class SomeClass(ParentClass):

    def __init__(*args, **kwargs):
        super().__init__(*args, **kwargs)
        # Do some other child class stuff
```

File: backend/hardware/reagentdispenser/base.py [hunk 4]
```diff
@@ -33,3 +33,31 @@ def getPumpSpeedLimits(pumpId):
                     Maximum speed the pump can dispense in ml/s
         """
         pass
+
+    @classmethod
+    def grblWrite(cls, grblSer, command, retries=3):
```
Comment (fall-byrd) [inline backend/hardware/reagentdispenser/base.py:38]: The implementation of this seemed to be the same from what I saw in both peristalticpump.py and syringepump.py so I moved it into their parent class.

**Review by fall-byrd — COMMENTED** [review]
Let me know if these comments were helpful or not, just trying to add some extra context :)
